test/WebSites/RazorPageExecutionInstrumentationWebSite/Startup.cs

```diff
@@ -22,24 +20,28 @@ public void ConfigureServices(IServiceCollection services)
 
             // Add MVC services to the services container.
             services.AddMvc();
-
-            // Make instrumentation data available in views.
-            services.AddScoped<TestPageExecutionListenerFeature, TestPageExecutionListenerFeature>();
-            services.AddScoped<IHoldInstrumentationData>(
-                provider => provider.GetRequiredService<TestPageExecutionListenerFeature>().Holder);
         }
 
         public void Configure(IApplicationBuilder app)
         {
+            var listener = new RazorPageDiagnosticListener();
+            var diagnosticSource = app.ApplicationServices.GetRequiredService<DiagnosticListener>();
+            diagnosticSource.SubscribeWithAdapter(listener);
```

Comment:
We don't have more than one tests here - so it really doesn't matter. We could fix it if we ever add more. `DiagnosticListener` derives from `DiagnosticSource` and it would have been confusing to have two `listeners` here.